File: src/main/java/tn/finhub/model/WalletTransaction.java
package tn.finhub.model;

import java.math.BigDecimal;
import java.time.LocalDateTime;

public class WalletTransaction {

    private int id;
    private int walletId;
<<<<<<< HEAD
    private String type; // CREDIT | DEBIT | HOLD | RELEASE
=======
    private String type;       // CREDIT | DEBIT | HOLD | RELEASE
>>>>>>> 3239865d261585c607c2f3379522c60b1fede853
    private BigDecimal amount;
    private String reference;
    private LocalDateTime createdAt;

<<<<<<< HEAD
    private String prevHash;
    private String txHash;

    public WalletTransaction() {
    }

    public WalletTransaction(int id, int walletId, String type,
            BigDecimal amount, String reference,
            String prevHash, String txHash,
            LocalDateTime createdAt) {
=======
    public WalletTransaction() {}

    public WalletTransaction(int id, int walletId, String type,
                             BigDecimal  amount, String reference,
                             LocalDateTime createdAt) {
>>>>>>> 3239865d261585c607c2f3379522c60b1fede853
        this.id = id;
        this.walletId = walletId;
        this.type = type;
        this.amount = amount;
        this.reference = reference;
<<<<<<< HEAD
        this.prevHash = prevHash;
        this.txHash = txHash;
=======
>>>>>>> 3239865d261585c607c2f3379522c60b1fede853
        this.createdAt = createdAt;
    }

    // Getters
    public int getId() {
        return id;
    }

    public int getWalletId() {
        return walletId;
    }

    public String getType() {
        return type;
    }

<<<<<<< HEAD
    public BigDecimal getAmount() {
=======
    public BigDecimal  getAmount() {
>>>>>>> 3239865d261585c607c2f3379522c60b1fede853
        return amount;
    }

    public String getReference() {
        return reference;
    }

<<<<<<< HEAD
    public String getPrevHash() {
        return prevHash;
    }

    public String getTxHash() {
        return txHash;
    }

=======
>>>>>>> 3239865d261585c607c2f3379522c60b1fede853
    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    // Setters
    public void setId(int id) {
        this.id = id;
    }

    public void setWalletId(int walletId) {
        this.walletId = walletId;
    }

    public void setType(String type) {
        this.type = type;
    }

<<<<<<< HEAD
    public void setAmount(BigDecimal amount) {
=======
    public void setAmount(BigDecimal  amount) {
>>>>>>> 3239865d261585c607c2f3379522c60b1fede853
        this.amount = amount;
    }

    public void setReference(String reference) {
        this.reference = reference;
    }

<<<<<<< HEAD
    public void setPrevHash(String prevHash) {
        this.prevHash = prevHash;
    }

    public void setTxHash(String txHash) {
        this.txHash = txHash;
    }

=======
>>>>>>> 3239865d261585c607c2f3379522c60b1fede853
    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
}

File: OneDrive/Desktop/FINHUB-WEB-main/src/Entity/WalletTransaction.php
<?php

namespace App\Entity;

use App\Repository\WalletTransactionRepository;
use Doctrine\DBAL\Types\Types;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: WalletTransactionRepository::class)]
#[ORM\Table(name: 'wallet_transactions')]
class WalletTransaction
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\ManyToOne(inversedBy: 'transactions')]
    #[ORM\JoinColumn(name: 'wallet_id', nullable: false)]
    private ?Wallet $wallet = null;

    #[ORM\Column(length: 20)]
    private ?string $type = null;

    #[ORM\Column(type: Types::DECIMAL, precision: 15, scale: 3)]
    private ?string $amount = null;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $reference = null;

    #[ORM\Column(name: 'prev_hash', length: 64, nullable: true)]
    private ?string $prevHash = null;

    #[ORM\Column(name: 'tx_hash', length: 64)]
    private ?string $txHash = null;

    #[ORM\Column(options: ['default' => 'CURRENT_TIMESTAMP'])]
    private ?\DateTimeImmutable $created_at = null;

    public function __construct()
    {
        $this->created_at = new \DateTimeImmutable();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getWallet(): ?Wallet
    {
        return $this->wallet;
    }

    public function setWallet(?Wallet $wallet): static
    {
        $this->wallet = $wallet;

        return $this;
    }

    public function getType(): ?string
    {
        return $this->type;
    }

    public function setType(string $type): static
    {
        $this->type = $type;

        return $this;
    }

    public function getAmount(): ?string
    {
        return $this->amount;
    }

    public function setAmount(string $amount): static
    {
        $this->amount = $amount;

        return $this;
    }

    public function getReference(): ?string
    {
        return $this->reference;
    }

    public function setReference(?string $reference): static
    {
        $this->reference = $reference;

        return $this;
    }

    public function getPrevHash(): ?string
    {
        return $this->prevHash;
    }

    public function setPrevHash(?string $prevHash): static
    {
        $this->prevHash = $prevHash;

        return $this;
    }

    public function getTxHash(): ?string
    {
        return $this->txHash;
    }

    public function setTxHash(string $txHash): static
    {
        $this->txHash = $txHash;

        return $this;
    }

    public function getCreatedAt(): ?\DateTimeImmutable
    {
        return $this->created_at;
    }

    public function setCreatedAt(\DateTimeImmutable $created_at): static
    {
        $this->created_at = $created_at;

        return $this;
    }
}
